Support BlurStyle values in BoxShadow (#52999)

deepak-penaganti opened the pull request:
*`BoxShadow` does not support all the `BlurStyle`s*

*Adds support for shadow types as mentioned in issue #52999*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

deepak-penaganti committed:
Support BlurStyle values in BoxShadow (#52999)

deepak-penaganti committed:
Support BlurStyle values in BoxShadow (#52999)

deepak-penaganti commented:
@googlebot I signed it!

HansMuller commented:
This PR will need tests.

@GaryQian - not sure who the right reviewer would be.

Piinks commented:
(triage) @deepak-penaganti would you like to continue with this change and add tests?

deepak-penaganti commented:
> (triage) @deepak-penaganti would you like to continue with this change and add tests?

Will add tests

deepak-penaganti committed:
Merge branch 'flutter:master' into blurstyle-for-boxshadow

deepak-penaganti committed:
Add tests for BoxShadow with blurStyle

deepak-penaganti committed:
Add tests for BoxShadow with blurStyle

deepak-penaganti commented:
@Piinks Added tests for the change

deepak-penaganti committed:
Remove unintended spaces

deepak-penaganti committed:
Remove unintended spaces

engine-flutter-autoroll committed:
Roll Engine from 98dc750f1c83 to 46178d6cdbcb (2 revisions) (#85982)

asashour committed:
Fix typo in comment (#85938)

deepak-penaganti committed:
Format space

deepak-penaganti committed:
Format space

deepak-penaganti committed:
More spaces removed

deepak-penaganti committed:
More spaces removed

asashour committed:
Fix EditableText when a controller listener changes values (#85956)

engine-flutter-autoroll committed:
Roll Engine from 46178d6cdbcb to 82015dfbd3d7 (3 revisions) (#85988)

engine-flutter-autoroll committed:
Roll Engine from 82015dfbd3d7 to 60eea42346c3 (3 revisions) (#85991)

Commit:
Update devtools_launcher.dart (#85995)

GaryQian commented on packages/flutter/lib/src/painting/box_shadow.dart:
```diff
@@ -36,11 +36,15 @@ class BoxShadow extends ui.Shadow {
   /// The amount the box should be inflated prior to applying the blur.
   final double spreadRadius;
 
+  /// The style(inner/normal/outer/solid) of the blur.
```
Let's keep the formatting/doc style consistent with the rest of Flutter. How about:

```
/// The [BlurStyle] to use for this shadow.
///
/// Defaults to [BlurStyle.normal].
```

deepak-penaganti replied:
Sure thing @GaryQian

ardera committed:
[custom-devices] general improvements, add custom-devices subcommand, better error handling (#82043)

engine-flutter-autoroll committed:
1626b1060 [video_player] Update to ExoPlayer 2.14.1 (flutter/plugins#4141) (#86022)

engine-flutter-autoroll committed:
1bce28c94 Roll Fuchsia Mac SDK from FYI3wC-M1... to YKiGmmdKT... (flutter/engine#27198) (#86024)

engine-flutter-autoroll committed:
b16697ca8 [flutter_plugin_tools] Migrate 'test' to new base command (flutter/plugins#4133) (#86026)

engine-flutter-autoroll committed:
Roll Engine from 1bce28c946a7 to dd17c0883ec0 (3 revisions) (#86027)

Commit:
[flutter_tools] revert change to SETLOCAL ENABLEDELAYEDEXPANSION (#86000)

engine-flutter-autoroll committed:
Roll Engine from dd17c0883ec0 to 84c62c657fda (6 revisions) (#86032)

engine-flutter-autoroll committed:
Roll Engine from 84c62c657fda to a1468f9b2495 (3 revisions) (#86036)

dnfield committed:
Assert disposed layers are not reused (#85798)

werainkhatri committed:
pass `style` argument to super of `IntProperty` (#85414)

Piinks committed:
[Framework] Support for Android Fullscreen Modes (#81303)

christopherfujino committed:
[flutter_conductor] support pushing local changes to remote (#85797)

engine-flutter-autoroll committed:
3f940429d [video_player] Fixed HLS Streams on iOS (flutter/plugins#3360) (#86053)

engine-flutter-autoroll committed:
6dd1c4658 [flutter_plugin_tools] Work around banner in drive-examples (flutter/plugins#4142) (#86059)

engine-flutter-autoroll committed:
Roll Engine from a1468f9b2495 to 00d9bec84191 (14 revisions) (#86068)

a14n committed:
code formatting (#85783)

engine-flutter-autoroll committed:
d0ac2f7c1 [various] Prepare plugin repo for binding API improvements (flutter/plugins#4137) (#86070)

engine-flutter-autoroll committed:
Roll Engine from 00d9bec84191 to 99a290467515 (2 revisions) (#86074)

engine-flutter-autoroll committed:
744730cd9 [plugin_platform_interface] Fix README broken link (flutter/plugins#4143) (#86077)

engine-flutter-autoroll committed:
Roll Engine from 99a290467515 to c989ae17b7c4 (3 revisions) (#86078)

engine-flutter-autoroll committed:
221c82954 Roll Skia from bc26cfc9a54a to 744c6a1b1992 (1 revision) (flutter/engine#27239) (#86084)

engine-flutter-autoroll committed:
c986b344c Roll Skia from 744c6a1b1992 to d471e7c36dd4 (4 revisions) (flutter/engine#27242) (#86086)

engine-flutter-autoroll committed:
e2541ca01 [camera] android-rework part 8: Supporting modules for final implementation (flutter/plugins#4054) (#86091)

engine-flutter-autoroll committed:
c4dde0c53 Roll Skia from d471e7c36dd4 to ae7f7edd499b (1 revision) (flutter/engine#27243) (#86092)

engine-flutter-autoroll committed:
fa036005b [in_app_purchase] Add documentation for price change confirmations (flutter/plugins#4092) (#86097)

deepak-penaganti committed:
Address review comments

deepak-penaganti committed:
Address review comments

engine-flutter-autoroll committed:
e1b4ba422 [in_app_purchase] Add currencySymbol to ProductDetails (flutter/plugins#4115) (#86098)

engine-flutter-autoroll committed:
b496703a3 Roll Skia from ae7f7edd499b to a2d6890c5b9b (1 revision) (flutter/engine#27244) (#86103)

chunhtai committed:
Fix ci.yaml and testowners format (#85565)

bparrishMines committed:
Create flag to enable/disable FlutterView render surface conversion  (#85648)

engine-flutter-autoroll committed:
6f00ff26d Roll Skia from a2d6890c5b9b to e822837a1322 (4 revisions) (flutter/engine#27246) (#86106)

engine-flutter-autoroll committed:
0b88ffb57 [video_player] Pause video when it completes (flutter/plugins#3727) (#86107)

jacob314 committed:
Switch drive_service from using deprecated version of --record-memory-profile flag. (#86122)

rickimaru18 committed:
Changing SnackBar's default vertical padding (#78133)

Piinks committed:
Remove TextTheme deprecations (#83924)

asashour committed:
[flutter_tools] show only supported sub commands (#85125)

zanderso committed:
[flutter_tool] Pin DevTools to 2.4.0 (#86124)

mghalayini committed:
Wrap FilterChip inside a Material widget (#85825)

asashour committed:
Add newline at end of file (#86008)

christopherfujino committed:
Migrate gen_localizations to null-safety (#86051)

christopherfujino committed:
[flutter_conductor] migrate conductor to null-safety (#86117)

Hixie committed:
Prevent merge conflicts when running customer tests. (#86063)

zanderso committed:
Revert "[flutter_tools] show only supported sub commands (#85125)" (#86131)

This reverts commit 84e576786c3dc5a839134b4862177c24f9030c85.

engine-flutter-autoroll committed:
200a6ec1d [in_app_purchase] Fix crash when retrieveReceiptWithError gives an error. (flutter/plugins#4138) (#86111)

engine-flutter-autoroll committed:
Roll Engine from 6f00ff26d767 to 9aa280e35589 (4 revisions) (#86115)

engine-flutter-autoroll committed:
Roll Plugins from 200a6ec1dbb3 to 06fc7d105446 (3 revisions) (#86132)

engine-flutter-autoroll committed:
Roll Engine from 9aa280e35589 to 7556ed4d7cb3 (8 revisions) (#86133)

engine-flutter-autoroll committed:
Roll Engine from 7556ed4d7cb3 to b2146208bb62 (3 revisions) (#86135)

engine-flutter-autoroll committed:
Roll Engine from b2146208bb62 to 6dcdb003001c (4 revisions) (#86146)

engine-flutter-autoroll committed:
dc0672409 Roll Skia from b1fd64e82efd to c2f5b311ceac (3 revisions) (flutter/engine#27269) (#86148)

engine-flutter-autoroll committed:
53d1a53ac Roll Dart SDK from eb6ae1c165fc to ae01d698532b (2 revisions) (flutter/engine#27270) (#86151)

dnfield committed:
Doc typo in layer.dart (#86033)

engine-flutter-autoroll committed:
40373cdcb Reland webp 1.2.0 (flutter/engine#27272) (#86155)

engine-flutter-autoroll committed:
dae2b2652 Roll Skia from c2f5b311ceac to 39122d445939 (1 revision) (flutter/engine#27271) (#86159)

engine-flutter-autoroll committed:
Roll Plugins from 06fc7d105446 to 59e16a556e27 (2 revisions) (#86160)

engine-flutter-autoroll committed:
Roll Engine from dae2b2652a4a to 024cc455fe9a (3 revisions) (#86162)

renyou committed:
Revert "Remove TextTheme deprecations (#83924)" (#86171)

This reverts commit 501474de994335e994c6a3314048f50b9abdfb01.

engine-flutter-autoroll committed:
2dfb40061 Roll Dart SDK from ae01d698532b to a8c4933bf92e (2 revisions) (flutter/engine#27279) (#86175)

Commit:
[flutter_tools] fix -n workaround (#86142)

engine-flutter-autoroll committed:
a011b374d Roll Skia from f5bd4e4b9f6b to beb2fbf0e66a (1 revision) (flutter/engine#27280) (#86178)

asashour committed:
Add .gitignore (#85017)

engine-flutter-autoroll committed:
1c24b5050 Use the prebuilt SDK's VM to run its frontend_server (flutter/engine#27274) (#86183)

asashour committed:
Fix typo in test name (#86083)

engine-flutter-autoroll committed:
6065bd001 [quick_actions] Add const constructor (flutter/plugins#4131) (#86185)

justinmc committed:
Dont reset the composing region when selection changes (#85799)

Fixes a bug on Android Gboard where the handle would disappear when dragged within a word.

engine-flutter-autoroll committed:
05884d394 Roll Skia from beb2fbf0e66a to 8573dab150a7 (4 revisions) (flutter/engine#27281) (#86189)

Commit:
[flutter] tab navigation does not notify the focus scope node (#86141)

Hixie committed:
Make the customer testing shard more verbose (#86143)

HansMuller committed:
Add a "troubleshooting" section to the AppBar API doc (#86188)

engine-flutter-autoroll committed:
Roll Engine from 05884d394cf0 to c373f704c367 (2 revisions) (#86194)

engine-flutter-autoroll committed:
Roll Engine from c373f704c367 to 0156ecd2e5bb (2 revisions) (#86195)

gspencergoog committed:
Add CallbackShortcuts widget (#86045)

yjbanov committed:
[web] move e2e tests from flutter/engine to flutter/flutter (#86119)

* [web] move e2e tests from flutter/engine to flutter/flutter
* flutter update-packages --force-upgrade

engine-flutter-autoroll committed:
fbb5e1383 [web] Librarify html renderer files (flutter/engine#27232) (#86201)

engine-flutter-autoroll committed:
8a966ebfd [various] Prepare plugin repo for binding API improvements (flutter/plugins#4148) (#86203)

engine-flutter-autoroll committed:
Roll Engine from fbb5e1383031 to 7482c6864a19 (5 revisions) (#86205)

engine-flutter-autoroll committed:
731b93d52 [image_picker_for_web] Migrate image_picker to package:cross_file (flutter/plugins#4083) (#86207)

engine-flutter-autoroll committed:
Roll Engine from 7482c6864a19 to 3ee0740f9a57 (2 revisions) (#86208)

engine-flutter-autoroll committed:
1199f131d [flutter_plugin_tools] Make unit tests pass on Windows (flutter/plugins#4149) (#86210)

engine-flutter-autoroll committed:
36271dc62 Roll Dart SDK from 83a559e0a9bd to ab8bac9fd542 (1 revision) (flutter/engine#27298) (#86212)

engine-flutter-autoroll committed:
0a892d9ec Roll Dart SDK from ab8bac9fd542 to 812bf07cf737 (1 revision) (flutter/engine#27300) (#86215)

engine-flutter-autoroll committed:
ac9142c96 encode DPR for shadows into DisplayList (flutter/engine#27289) (#86220)

engine-flutter-autoroll committed:
8e6868972 Roll Fuchsia Linux SDK from 7q7h2ljWo... to C6i1FgtOK... (flutter/engine#27304) (#86224)

engine-flutter-autoroll committed:
c341a03bf Roll Fuchsia Mac SDK from MkrnXce9t... to MKuIBsUt-... (flutter/engine#27305) (#86233)

engine-flutter-autoroll committed:
5f445d7c0 Roll Dart SDK from 812bf07cf737 to 40695d910591 (1 revision) (flutter/engine#27306) (#86239)

engine-flutter-autoroll committed:
Roll Engine from 5f445d7c0811 to 8bb2641d37d4 (2 revisions) (#86241)

engine-flutter-autoroll committed:
06b7d1f8c Roll Fuchsia Linux SDK from C6i1FgtOK... to FV0wVt4TU... (flutter/engine#27310) (#86242)

engine-flutter-autoroll committed:
Roll Engine from 06b7d1f8c6c3 to ed25e8f01efe (2 revisions) (#86256)

darrenaustin committed:
Migrate core devicelab framework to null safety. (#85993)

Migrate core devicelab framework to null safety.

darrenaustin committed:
Migrate core devicelab tasks a-f null safety. (#85996)

darrenaustin committed:
Migrate core devicelab tasks f-i null safety. (#85997)

zanderso committed:
Revert "Migrate core devicelab tasks f-i null safety. (#85997)" (#86267)

This reverts commit d4afa28957a499b4e773efa24e824dde1439d8fa.

darrenaustin committed:
Revert "Migrate core devicelab tasks a-f null safety. (#85996)" (#86268)

This reverts commit 9d3563a8eaba076f7a4b042764d8abf7e7e0467f.

darrenaustin committed:
Revert "Migrate core devicelab framework to null safety. (#85993)" (#86269)

This reverts commit 2175e64e4fdb387c5f01242c9a82779d3423e531.

engine-flutter-autoroll committed:
Roll Engine from ed25e8f01efe to 0e2b9bebfb8d (2 revisions) (#86273)

engine-flutter-autoroll committed:
7851449b4 Roll Fuchsia Mac SDK from gySDhb-Ea... to DM-zc7Snb... (flutter/engine#27322) (#86277)

deepak-penaganti commented:
@iapicca @TahaTesser @darshankawar Please do get a look at this PR and assign [this](https://github.com/flutter/flutter/issues/52999) related issue

Commit:
[flutter_tools] switch web integration tests to use html (#85325)

payam-zahedi committed:
borderRadius param updated in copyWith functions (#85822)

Commit:
[flutter_tools] let the logger know about machine mode (#86116)

engine-flutter-autoroll committed:
Roll Engine from 7851449b4d33 to 26b430373207 (2 revisions) (#86301)

michaellee8 committed:
[flutter_tools] Add support for launching fuchsia app using session_control (#85752)

HansMuller committed:
AppBar.backwardsCompatibility now default false, deprecated (#86198)

Hixie committed:
Run "pub global activate devtools" before overall_experience_test (#85812)

engine-flutter-autoroll committed:
Roll Engine from 26b430373207 to 5a3945cdcbcd (6 revisions) (#86319)

Piinks committed:
Revert "Revert "Remove TextTheme deprecations (#83924)" (#86171)" (#86318)

engine-flutter-autoroll committed:
Roll Engine from 5a3945cdcbcd to a984a833baaf (2 revisions) (#86322)

engine-flutter-autoroll committed:
d68f26313 [flutter_plugin_tools] Support format on Windows (flutter/plugins#4150) (#86327)

GaryQian commented:
Ok, so the existing code all looks great, but since much of the rendering of the shadow happens in the engine, we should ensure all of the mechanisms that pass the style down are all functional and tested.

A great way to do this is a golden which can ensure the final drawn result has the correct shadow and visually looks correct.

Try something like:

```
testWidgets('BoxShadow BoxStyle.solid', (WidgetTester tester) async {
    final Key key = UniqueKey();
    await tester.pumpWidget(
      Center(
        child: RepaintBoundary(
          key: key,
          child: Container(
            color: Colors.white,
            width: 50,
            height: 50,
            child: Center(
              child: Container(
                decoration: BoxDecoration(
                  boxShadow: <BoxShadow>[BoxShadow(blurRadius: 3.0, blurStyle: BlurStyle.inner)],
                ),
                width: 10,
                height: 10,
              ),
            ),
          ),
        ),
      ),
    );

    await expectLater(
      find.byKey(key),
      matchesGoldenFile('boxShadow.boxStyle.inner.0.0.png'),
    );
  });
```

We can then use SkiaGold in the presubmit checks to commit the golden png images.

deepak-penaganti committed:
Add support for SkiaGold presubmit check

deepak-penaganti committed:
Add support for SkiaGold presubmit check

deepak-penaganti committed:
Merge branch 'blurstyle-for-boxshadow' of https://github.com/deepak-penaganti/flutter into blurstyle-for-boxshadow

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 9.
View them at https://flutter-gold.skia.org/cl/github/84733

deepak-penaganti commented:
@GaryQian Not sure why **_customer_testing-linux_** keeps failing 🤔

engine-flutter-autoroll committed:
Roll Engine from a984a833baaf to 0bf8bfdf1796 (5 revisions) (#86350)

HansMuller committed:
Additional ListTile API doc - Material widget dependency (#86355)

Piinks committed:
Dart Fixes for clipBehavior breaks (#86323)

engine-flutter-autoroll committed:
Roll Engine from 0bf8bfdf1796 to 07d04041d58e (3 revisions) (#86356)

KristofMerey committed:
Fix documentation of equality functions. (#84847)

The first line of the documentation of equality functions listEquals, mapEquals, and setEquals claimed that they are deep equality functions. A later paragraph explained that they check for deep equality only if the elements, which are collections, implement the equality operator to do so. That is almost never the case.

The first line of the documentation was changed to element-by-element equality, and a reference was added to the DeepCollectionEquality class.

Commit:
Revert "[flutter_tools] let the logger know about machine mode (#86116)" (#86363)

This reverts commit 53d8cba3ec8d75288107c5404472809eab61abe8.

engine-flutter-autoroll committed:
ce371a6e7 [webview_flutter] Fix broken keyboard issue link (flutter/plugins#3266) (#86358)

dependabot[bot] committed:
Bump addressable from 2.7.0 to 2.8.0 in /dev/ci/mac (#86329)

werainkhatri committed:
Added rethrowError to FutureBuilder (#84308)

darrenaustin committed:
Migrate devicelab framework code to null safety. (#86325)

(Attempt to reland #85993)

engine-flutter-autoroll committed:
Roll Engine from 07d04041d58e to c74f25f4cb0a (8 revisions) (#86366)

Hixie committed:
Clean up the bindings APIs (#83843)

Hixie committed:
Revert "Clean up the bindings APIs (#83843)" (#86386)

This reverts commit e2490f2906060d0aaecbbe34a9240ca3d54ef998.

darrenaustin committed:
Migrate devicelab tasks i-z to null safety. (#85998)

darrenaustin committed:
Migrate the first 3rd of the devicelab tasks to null safety. (#86374)

Commit:
[flutter] prevent errant text field clicks from losing focus (#86041)

HansMuller committed:
Added AppBarTheme shape property (#86372)

Commit:
[ci.yaml] Nest tags under properties (#86364)

Hixie committed:
Add more debugging logs to overall_experience_test (#86368)

engine-flutter-autoroll committed:
Roll Engine from c74f25f4cb0a to 57720b2f3679 (2 revisions) (#86373)

darrenaustin committed:
Fixed a cast issue with host_mode_tests.dart. (#86392)

engine-flutter-autoroll committed:
fbb4b3a85 [flutter_plugin_tools] Improve license-check output (flutter/plugins#4154) (#86380)

Hixie committed:
Clean up the bindings APIs (#86388)

engine-flutter-autoroll committed:
Roll Engine from 57720b2f3679 to bde2f528888b (8 revisions) (#86390)

zanderso committed:
Revert "Clean up the bindings APIs (#86388)" (#86404)

This reverts commit 31de052e3fddb4909ff88172a6937db7f3dc49db.

whesse committed:
Always write devicelab test results to a file if the resultsPath option is present (#86393)

Some runs of DeviceLab performance tests are run without a gitBranch option, but
still must save their results to a local file.

darrenaustin committed:
Migrate core devicelab tasks f-i null safety. (#86378)

engine-flutter-autoroll committed:
Roll Engine from bde2f528888b to 093c4ed1a3d7 (3 revisions) (#86406)

yjbanov committed:
[web] use resident resident runner in flutter drive (#86381)

* [web] use resident resident runner in flutter drive

Piinks committed:
Add fixes for ThemeData (#86369)

dkwingsmt committed:
New scheme for keyboard logical key ID (#85121)

This PR updates the ID used by logical keyboard keys.

The logical key ID is still composed of 2 parts: 32 bits of value, and 8 bits of plane. But the assignment of planes has been drastically changed. HID plane is removed, and unprintable plane and Flutter plane are added. This is to reflect the new generation method for logical key IDs. Now keys that are defined by Flutter but not by dom_key_data are placed into the Flutter plane, including numpad keys, sided modifier keys, and gamepad keys. The values for platform planes have also been adjusted.

The generation script and README have been updated accordingly as well.

A new file, test_utils/key_codes.h is now generated to assist engine unit testing.

All lists generated by the script are now sorted by the key.

Juneezee committed:
build: update dependencies (#86433)

guidezpl committed:
[Fonts] Fix icons sorting (#86434)

* Fix sorting

* Update icons.dart

Hixie committed:
Clean up the bindings APIs (#86438)

goderbauer commented:
> @GaryQian Not sure why customer_testing-linux keeps failing 🤔

Looks like a flake. I've restarted the test. If it continues to fail, you may have to rebase this branch on top of the latest master.

keyonghan committed:
Rerun devicelab task from test runner (#86394)

Commit:
[flutter_tools] support trailing args (#86431)

GaryQian commented on packages/flutter/test/painting/box_painter_test.dart:
```diff
@@ -107,7 +109,70 @@ void main() {
+    expect(const BoxShadow(blurRadius: 4.0, blurStyle: BlurStyle.solid).toString(), equals('BoxShadow(Color(0xff000000), Offset(0.0, 0.0), 4.0, 0.0), BlurStyle.solid'));
+  });
+
+  testWidgets('BoxShadow BoxStyle.solid', (WidgetTester tester) async {
```
One last thing, while you are at it, it should be easy to copypaste this test a few times to cover all of the different BlurStyles!

deepak-penaganti replied:
Done 😃

GaryQian commented:
Customer_testing-linux still seems to be failing, why don't you try rebasing? I don't think this change should cause any failures in that test shard.

`git pull upstream master --rebase`

guidezpl committed:
[Fonts] Use exact matching for icon identifier rewrites (#86464)

* Fix sorting

* Update icons.dart

* [Fonts] Use instead matching instead of prefix

* Add new

Commit:
Add an Interactive Example for PhysicalShape (#86423)

renyou committed:
Revert "Clean up the bindings APIs (#86438)" (#86484)

This reverts commit d056500bfe6a712cdace450d06c4a8fb2290e63a.

zanderso committed:
Add iOS benchmarks for sksl warmup (#86460)

deepak-penaganti committed:
Merge branch 'blurstyle-for-boxshadow' of https://github.com/deepak-penaganti/flutter into blurstyle-for-boxshadow

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 169.
View them at https://flutter-gold.skia.org/cl/github/84733

deepak-penaganti committed:
Add more tests

skia-gold commented:
Gold has detected about 2 new digest(s) on patchset 170.
View them at https://flutter-gold.skia.org/cl/github/84733

christopherfujino commented:
The diff on this PR looks wrong (it's so big I can't load it in my browser), likely due to either an incorrect branch point (not from master), or an incorrect merge (not from master). This PR cannot be landed as is.

christopherfujino reviewed (changes requested):
Please fix the diff on this PR (it is too large)

deepak-penaganti commented:
@christopherfujino I hear you, not sure how merge went wrong. Please do check this [PR#86504](https://github.com/flutter/flutter/pull/86504)

deepak-penaganti commented:
We can close this in favour of #86504

christopherfujino commented:
closing in favor of https://github.com/flutter/flutter/pull/86504
